### ISSUE TITLE
Push notifications

### ISSUE THREAD

**Issue by pminten** [exercism/rivulet#14]
It seems that the now removed email feature was popular with people who appreciated the push into their personal queue (i.e. their email inbox). It would be good if rivulet has some kind of delivery mechanism that supports the same use case as email.

I think the main requirements for this are:

* It integrates into a user's existing workflow. The user does not need to check a new system. Email is actually perfect for this since it's ubiquitous. Something like a Twitter DM would probably work too, though not everyone's on twitter.
* It does not cause the user to feel spammed. In the case of email this means no separate mails for each message unless the user explicitly wants it.

**Comment by kytrinyx** [comment]
I agree. With this feature we will already have everything in place to do granular personalised configuration, and the notification code will be isolated.
